hammers/query.py

```diff
@@ -284,6 +284,52 @@ def update_orphan_resource_providers(db):
     '''
     return db.query(sql, no_rows=True)
 
+@query
+def get_advanced_reservations(db):
+    """Get all advanced reservations created by any user"""
+    sql = '''\
+    SELECT lu.name AS user_name, u.extra AS user_extra, l.name AS lease_name, l.id AS lease_id, p.name AS project_name, l.start_date AS start_date 
+    FROM blazar.leases AS l
+    JOIN keystone.user AS u ON l.user_id = u.id 
+    JOIN keystone.local_user AS lu ON u.id = lu.user_id 
+    JOIN keystone.project AS p ON l.project_id = p.id 
+    WHERE l.start_date > UTC_TIMESTAMP()
+    '''
+    
+    return db.query(sql, limit=None)
+
+@query
+def get_idle_leases(db, hours):
+    """
+    Get all leases that have been idled for more than certain hours.
+    An idled lease is defined as an ongoing lease without active instance and without openstack operation.
```

Comment:
s/idled/idle/

Reply:
What does without openstack operation mean?

Reply:
such as prolong lease and change lease name, i.e. lease-update.

Reply:
Then I would say more explicitly: `without lease update operation`